libraries/timeslider/src/components/TimeSlider/TimeSlider.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { ComponentType, ReactElement, useCallback } from "react";
+import { LayoutElementProperties } from "@vertigis/web/components";
+import EsriTimeSlider from "@arcgis/core/widgets/TimeSlider";
+import TimeSliderModel from "./TimeSliderModel";
+import {
+    createEsriMapWidget,
+    MapWidgetProps,
+} from "@vertigis/web/ui/esriUtils";
+import "./TimeSlider.css";
+import { useWatch, useWatchAndRerender } from "@vertigis/web/ui";
+import WebMap from "@arcgis/core/WebMap";
+
+export type TimeSliderProps = MapWidgetProps<TimeSliderModel, EsriTimeSlider>;
+const TimeSliderWrapper: ComponentType<TimeSliderProps> = createEsriMapWidget<
+    TimeSliderModel,
+    EsriTimeSlider
+>(EsriTimeSlider, undefined, true);
+
+const TimeSlider = (
+    props: LayoutElementProperties<TimeSliderModel>
+): ReactElement => {
+    // A unique DOM ID to be used for a11y purposes.
+    const model: TimeSliderModel = props.model;
+
+    // Any time the Time Slider model properties change, we need to re-render
+    // the component so our UI is consistent, as well as set the Esri widget
+    // configuration.
+    useWatchAndRerender(model, [
+        "layout",
+        "loop",
+        "playRate",
+        "mode",
+        "overrideStops",
+        "timeInterval",
+        "timeIntervalUnit",
+        "timeVisible",
+        "map.map",
+    ]);
+    useWatch(
+        model,
+        [
+            "layout",
+            "loop",
+            "playRate",
+            "mode",
+            "overrideStops",
+            "timeInterval",
+            "timeIntervalUnit",
+            "timeVisible",
+        ],
+        (newValue, _oldValue, propertyName) => {
+            if (model.widget) {
+                model[propertyName] = newValue;
```

Comment:
Am I missing something, or is this watch-handler setting a the same value on the model that effectively triggered the handler in the first place?

eg.
model.foo = "bar";
--> watchHandle("foo", undefined, "foo") // model["foo"] = "bar";

Reply:
The watch handlers have been removed from the latest updates.